src/services/ui/src/pages/detail/package-activity/hook.ts

```diff
@@ -0,0 +1,61 @@
+import { getAttachmentUrl } from "@/api";
+import JSZip from "jszip";
+import { saveAs } from "file-saver";
+import { useState } from "react";
+import { opensearch } from "shared-types";
+
+type Attachments = NonNullable<opensearch.changelog.Document["attachments"]>;
+
+export const useAttachmentService = (
+  props: Pick<opensearch.changelog.Document, "packageId">
+) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const onUrl = async (ATT: Attachments[number]) => {
+    setLoading(true);
+    try {
+      return await getAttachmentUrl(
+        props.packageId,
+        ATT.bucket,
+        ATT.key,
+        ATT.filename
+      );
+    } catch (e) {
+      const err = e instanceof Error ? e.message : "Failed download";
+      setError(err);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
Can we utilize `react-query` for this, and pass the export thru this hook?

Reply:
I completely overlooked that... great point out! 
Makes it much cleaner 🤟